crates/subspace-gateway/src/commands/http.rs

```diff
@@ -33,33 +34,48 @@ pub async fn run(run_options: HttpCommandOptions) -> anyhow::Result<()> {
     } = run_options;
 
     let (object_fetcher, mut dsn_node_runner) = initialize_object_fetcher(gateway_options).await?;
-    let dsn_fut = dsn_node_runner.run();
+    let dsn_fut = run_future_in_dedicated_thread(
+        move || async move { dsn_node_runner.run().await },
+        "gateway-networking".to_string(),
+    )?;
 
     let server_params = ServerParameters {
         object_fetcher,
         indexer_endpoint,
         http_endpoint: http_listen_on,
     };
-    let http_server_fut = start_server(server_params);
+    let http_server_handle = actix_web::rt::spawn(start_server(server_params));
```

Comment:
This future is not `Send` (can't be safely sent between threads), so we have to spawn it locally on this thread, and start another thread for the signal future (and any other independent futures).